extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
 		global $wgConsulUrl;
 
-		$esUrl = (new  Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, 'query', 'sjc' ))->getUrl( 'es' );
+		$esUrl = (new Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, 'query', 'sjc' ))->getUrl( 'es' );
 		$date = time() - ( $dayOffset * 24 * 60 * 60 );
 
 		return 'http://' . $esUrl . '/logstash-' . date( 'Y.m.d', $date ) . '/_search';
 	}
 
-	private function getUserIdByUsername( $userName ) {
+	private function getUserByUsername( $userName ) {
 		if ( empty( $userName ) ) {
-			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
 		}
 
 		$user = User::newFromName( $userName );
-		if ( !$user || empty( $user->getId() ) ) {
-			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+		if ( !$user ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
+		}
+
+		return $user;
+	}
+
+	private function getUserById( $userId ) {
+		if ( !$userId ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
+		}
+
+		// Look at cache first
+		if ( !empty( $this->users[ $userId ] ) ) {
+			return $this->users[$userId];
+		}
+
+		$user = User::newFromId( $userId );
+		if ( !$user ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
 		}
 
-		return $user->getId();
+		$this->users[ $user->getId() ] = $user;
+
+		return $user;
 	}
 
 	public function userLog() {
-		$userName = $this->getVal( 'username' );
+		list( $userName, $ipAddress ) = $this->readAndValidateInput();
 
 		$userLogRecords = [];
-		$displayedUserLogRecords = [];
-		$hasNoUserLogRecords = false;
 		$hasUserError = false;
 		$userId = null;
 		$userErrorMessage = null;
 
-		try {
-			$userId = $this->getUserIdByUsername( $userName );
-		} catch ( Exception $e ) {
-			$hasUserError = true;
-		}
+		if ( !empty( $userName ) ) {
+			try {
+				$userId = $this->getUserByUsername( $userName )->getId();
+			} catch ( \Exception $e ) {
+				$hasUserError = true;
+			}
 
-		if ( $hasUserError !== true ) {
-			$userLogRecords = $this->aggregateLogSearches( $userId, $userName );
-		}
+			if ( !$hasUserError ) {
+				$userLogRecords = $this->aggregateLogByUserId( $userId );
+			}
+
+			$this->logTableCaption = wfMessage( 'discussionslog-table-caption' )
+				->params( [ $userName, $userId ] )
+				->escaped();
+
+			$this->noUserLogRecordsMessage = wfMessage( 'discussionslog-no-mobile-activity-error' )
+				->params( $userName )
+				->escaped();
+
+			$this->userErrorMessage = wfMessage( self::NO_USER_MATCH_ERROR )->escaped();
+
+		} else if ( !empty( $ipAddress ) ) {
+			$userLogRecords = $this->aggregateLogByIpAddress( $ipAddress );
 
-		if ( count( $userLogRecords ) == 0 ) {
-			$hasNoUserLogRecords = true;
+			$this->logTableCaption = wfMessage( 'discussionslog-table-ip-caption' )
```

Comment:
Some of these response values are being set indirectly via member variables, and others are set explicitly inside of `$this->response->setValues(`. Can we set all of these these explicitly to be consistent?